### PR TITLE
Modify migrations scripts to remove shellJs

### LINKED ISSUE #1841: Switch `truffle migrate` to follow the ZOS example
**Is your feature request related to a problem? Please describe.**
We are currently using shell commands to interact with ZOS, however they just released a better way.

**Describe the solution you'd like**
Follow the example here: https://github.com/zeppelinos/zos/tree/master/examples/truffle-migrate

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by nfurfaro**
# Description

This PR removes the usage of `shell.js` in our migrations scripts. 

# Issues

Fixes #1841
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by hardlydifficult** [comment]
This is failing for me..

```
npx truffle migrate
...
Error: Returned values aren't valid, did it run Out of Gas?
```

Which is similar to what I saw with my draft.  Is this working for you?

**Comment by nfurfaro** [comment]
I get:
```
overlord @ ~/DAPPS/unlock/smart-contracts : n44o-issue-1841-remove-shell  => npx truffle migrate

Compiling your contracts...
===========================
> Everything is up to date, there is nothing to compile.

Network up to date.
```

**Comment by nfurfaro** [comment]
Note that this PR is only addressing the removal of `shell.js`

**Comment by hardlydifficult** [comment]
Just reverted my vm to base lubuntu and testing a clean install now.

File: smart-contracts/migrations/2_deploy_unlock.js [hunk 1]
```diff
@@ -1,21 +1,35 @@
...
-        ).code !== 0
-      ) {
-        throw new Error('Migration failed')
-      }
+      const {
+        network,
+        txParams
+      } = await ConfigVariablesInitializer.initNetworkConfiguration({
+        network: networkName,
+        // from: accounts[1]
```
Comment (hardlydifficult) [inline smart-contracts/migrations/2_deploy_unlock.js]: Could set this to `from: unlockOwner`

Comment (hardlydifficult) [inline smart-contracts/migrations/2_deploy_unlock.js]: Maybe just remove this comment

File: smart-contracts/migrations/3_create_proxy.js [hunk 2]
```diff
@@ -1,23 +1,46 @@
...
+// Load zos scripts and truffle wrapper function
+const { scripts, ConfigVariablesInitializer } = require('zos')
+const { create } = scripts
 
-module.exports = function deployProxy (deployer, network, accounts) {
+async function deploy (options, accounts) {
   // default account used by ganache
   const unlockOwner = accounts[0]
+
+  // Create an instance of MyContract, setting initial value to 42
```
Comment (hardlydifficult) [inline smart-contracts/migrations/3_create_proxy.js]: Please delete or update the comments here

File: smart-contracts/migrations/3_create_proxy.js [hunk 3]
```diff
@@ -1,23 +1,46 @@
...
+        initMethod: 'initialize',
+        initArgs: [unlockOwner]
+      },
+      options
+    )
+  )
+}
+
+module.exports = function (deployer, networkName, accounts) {
   // zos proxy admin can be any account other than accounts[0]
```
Comment (hardlydifficult) [inline smart-contracts/migrations/3_create_proxy.js]: This comment does not apply anymore.

Comment (hardlydifficult) [reply smart-contracts/migrations/3_create_proxy.js]: The admin can be any address now

Comment (nfurfaro) [reply smart-contracts/migrations/3_create_proxy.js]: Righ. I wanted to ask you about this.

File: smart-contracts/migrations/3_create_proxy.js (hunk 2 above)
Comment (nfurfaro) [reply smart-contracts/migrations/3_create_proxy.js]: Yup.

File: smart-contracts/migrations/2_deploy_unlock.js (hunk 1 above)
Comment (nfurfaro) [reply smart-contracts/migrations/2_deploy_unlock.js]: done.

**Comment by hardlydifficult** [comment]
Small typo in the last push: 3_create_proxy.js line 18

**Comment by nfurfaro** [comment]
I also had to fix the deploy function in the same script.

**Pull request merged**